pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,346 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from enum import IntFlag, Flag
+import time
+from pymeasure.adapters import VISAAdapter
+from pymeasure.instruments.validators import strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+# pylint: disable=W1203
+
+
+class NI_GPIB_232(VISAAdapter):
+    """Encapsulates the additional commands necessary
+       to communicate over a National Instruments GPIB-232CT Adapter,
```

Comment:
Why leave out the `CT` part of the model name in the class name? Is that some variant indicator? Or is that `CT-A`?

Also, following the CamelCase convention for class names, this should probably be `NIGPIB232CT`, but that might be unwieldy? Thoughts?

Reply:
As far as I oversee the list of N GPIB-RS232/485 adapters, there are:
- the NI GPIB-RS232 (which is still listed for sale on the NI website)
- the NI GPIB-232/485CT-A and
- the NI GPIB-232CV-A
- the NI GPIB-232CT (older device)

The 232/485CT-A can be used as GPIB master and slave device,
while the CV-A-models are only for converting a device with RS232 to a GPIB-slave or connecting to a single GPIB-device.
Detail can be found [here](https://www.apexwaves.com/pdfs/gpib-232ct-a-user-manual.pdf) 

The `-A` suffix could be omitted, but is printed on the device itself. 
I have found in the meantime a manual for the oolder 232CT version but still will have to compare the commands

For the naming, at the moment I would be ok with` NIGPIB232CT`.
But let me give you a more precive feedback, once I investigated the differewnces between CT-A and CT.

Reply:
> For the naming, at the moment I would be ok with` NIGPIB232CT`.
> But let me give you a more precive feedback, once I investigated the differewnces between CT-A and CT.

Sounds good. It might be good to add that useful info you found on the model differences to the source, either in a docstring or code comment (just so it's not lost).

Reply:
> But let me give you a more precive feedback, once I investigated the differewnces between CT-A and CT.

I went trought the S-mode description of the 232CT and 232/485CT-A, for the scope of the impelmentation both devices use the same commands and syntax.

updated docstring